Source/AwsCommonRuntimeKit/event-stream/EventStreamMessage.swift

```diff
@@ -47,90 +47,89 @@ public struct EventStreamMessage {
 
 extension EventStreamMessage {
     func addHeader(header: EventStreamHeader, rawHeaders: UnsafeMutablePointer<aws_array_list>) throws {
-        if header.name.count > EventStreamHeader.maxNameLength {
+        if header.name.utf8.count > EventStreamHeader.maxNameLength {
             throw CommonRunTimeError.crtError(
                 .init(
                     code: AWS_ERROR_EVENT_STREAM_MESSAGE_INVALID_HEADERS_LEN.rawValue))
         }
         let addCHeader: () throws -> Int32 = {
-            return try header.name.withCString { headerName in
-                switch header.value {
-                case .bool(let value):
-                    return aws_event_stream_add_bool_header(
-                        rawHeaders,
-                        headerName,
-                        UInt8(header.name.count),
-                        Int8(value.uintValue))
-                case .byte(let value):
-                    return aws_event_stream_add_byte_header(
-                        rawHeaders,
-                        headerName,
-                        UInt8(header.name.count),
-                        value)
-                case .int16(let value):
-                    return aws_event_stream_add_int16_header(
-                        rawHeaders,
-                        headerName,
-                        UInt8(header.name.count),
-                        value)
-                case .int32(let value):
-                    return aws_event_stream_add_int32_header(
+            let headerNameLength = UInt8(header.name.utf8.count)
```

Comment:
trivial: move this up, so we can do `if headerNameLength > EventStreamHeader.maxNameLength`
instead of needing to type the easy-to-mess-up `.utf8.count` multiple times